linux_os/guide/system/software/updating/ensure_fedora_gpgkey_installed/bash/shared.sh

```diff
@@ -25,14 +25,14 @@ if [ "${RPM_GPG_DIR_PERMS}" -le "755" ]
 then
   # If they are safe, try to obtain fingerprints from the key file
   # (to ensure there won't be e.g. CRC error).
-  IFS=$'\n' GPG_OUT=($(gpg --with-fingerprint "${REDHAT_RELEASE_KEY}" | grep 'Key fingerprint ='))
+  IFS=$'\n' GPG_OUT=($(gpg --with-fingerprint --with-colons "${REDHAT_RELEASE_KEY}" | grep '"^fpr' | cut -d ":" -f 10))
```

Comment:
Unfortunately, this is "abuse" of `IFS` - the correct way is to save `IFS` and then restore it. Setting `IFS` on the same line may create a false impression that it applies to that one line only, but in reality, it is overwritten from that place on (see [the bash video](https://redhat.bluejeans.com/playback/guid/NDg0NjkwNzgyMDozMTczMTctNjk4MDQ5OTYtNmM5Yi00ZTBmLWFhNDctNWJlZGJlMzJkZjFj) around 17. minute). Unsetting IFS may restore it to the original value, but it is [reported](https://stackoverflow.com/questions/39545837/unset-ifs-unexpected-behaviour) to be unsafe. So I suggest to have there an auxiliary `SAVE_IFS` variable.

Reply:
Thanks for the references, unfortunately I cannot see the video.